src/subdomains/generic/user/models/auth/auth.service.ts

```diff
@@ -97,8 +97,23 @@ export class AuthService {
   }
 
   // --- AUTH METHODS --- //
-  async authenticate(dto: CreateUserDto, userIp: string): Promise<AuthResponseDto> {
+  async authenticate(dto: CreateUserDto, userIp: string, userDataId: number): Promise<AuthResponseDto> {
     const existingUser = await this.userService.getUserByAddress(dto.address, { userData: true, wallet: true });
+    const userData = userDataId && (await this.userDataService.getUserData(userDataId, { users: true }));
+
+    if (userData) {
+      if (existingUser) {
+        if (existingUser.userData.id === userDataId)
+          return { accessToken: this.generateUserToken(existingUser, userIp) };
+        throw new ConflictException('User already exists');
+      }
+
+      const newUser = await this.userService.createUser({ userDetails: dto, userIp, userData });
+      await this.userDataService.addNewUser(userData, newUser);
+
```

Comment:
The `this.userDataService.addNewUser` step is actually not required 🤔  UserData connection is already saved in `createUser`